src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,158 @@
+package duke;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+
+public class Parser {
+    private boolean isDate(String time) {
+        try {
+            SimpleDateFormat date = new SimpleDateFormat("yyyy-MM-dd");
+            date.setLenient(false);
+            date.parse(time.trim());
+        } catch (ParseException e) {
+            return false;
+        }
+        return true;
+    }
+
+
+    /**
+     * Returns the date (MMM dd yyyy) of the specified time
+     * Time must be formatted as yyyy-MM-dd to be converted to date
+     * If time does not match the format, the original time will be returned
+     *
+     * @param time Time specified in the task
+     * @return Time in date format
+     */
+    public String convertDate(String time) {
+        String s = "";
+        try {
+            if (isDate(time)) {
+                Date d = new SimpleDateFormat("yyyy-MM-dd").parse(time);
+                s = new SimpleDateFormat("MMM dd yyyy").format(d);
+            } else {
+                s = time;
+            }
+        } catch (ParseException e) {
+            System.out.println("Please specify date in the form yyyy-MM-dd");
+        }
+        return s;
+    }
```

Comment:
Very good usage of abstraction in order to let one method take care of converting a date. I realise I should have done that as well haha. Good idea!